pkg/controller/daemon/daemon_controller.go

```diff
@@ -1537,7 +1538,9 @@ func getPodsWithoutNode(runningNodesList []*v1.Node, nodeToDaemonPods map[string
 	for n, pods := range nodeToDaemonPods {
 		if !isNodeRunning[n] {
 			for _, pod := range pods {
-				results = append(results, pod.Name)
+				if len(pod.Spec.NodeName) == 0 {
```

Comment:
LGTM overall; personally, I'd like to add a comment on the scenario that DaemonSet-Controller will delete Pod without NodeName, and PodGC will delete pod has NodeName.